delivery/client.py

```diff
@@ -111,28 +121,32 @@ def component_descriptor(
 
     def upload_metadata(
         self,
-        data: dso.model.ArtefactMetadata,
+        data: typing.Iterable[dso.model.ArtefactMetadata],
```

Comment:
stylistic nit: you might have declared data as an "arbitrary positional argument" (`*data`). that way, the API might be slightly more convenient to use